main.test.js

```diff
@@ -1,38 +1,233 @@
-
-// add jquery to global scope
 global.__DEV__ = true
 global.$ = require('jquery');
+global.d3 = require('d3');
+
+const puppeteer = require('puppeteer');
+const fs = require('fs');
 
-let main;
+describe('Opening viewTree.html', () => {
+  let contentHtml = fs.readFileSync('./viewtree.html', 'utf8');
+  let browser;
+  let page;
+  beforeEach(async () => {
+    browser = await puppeteer.launch();
+    page = await browser.newPage();  
+    // await page.setContent(contentHtml, { waitUntil: "networkidle0" });
+    // await page.addScriptTag({path: "main.js"});
+    await page.goto("http://127.0.0.1:5500/viewtree.html"); // do this to be able to call functions in main.js, might need to direct the browser at this url
+  });
+  afterEach(() => {
+    browser.close();
+  });
+
+  // tests to be used
+  // Run a specfic test: npm test -- -t  'test_name_matcher' --verbose
+
+  test('Page title should be Palmyra v2.4', async () => {
+    await expect(page.title()).resolves.toMatch('Palmyra v2.4');
+  });
 
-beforeAll(() => {
-  // DOM doesn't load before code runs,
-  // initialize html elements used in main.js
-  // (ids found in viewtree.html)
+  test('Body should contain class viewtree', async () => {
+    await page.$eval('body.viewtree', el => el.text);
+  });
   
-  // create element, add id, then append to document body
-  document.body.appendChild(
-    Object.assign(document.createElement('input'),{id:"filename"})
-  );
-  document.body.appendChild(
-    Object.assign(document.createElement('input'),{id:"configFile"})
-  );
-
-  // include main.js
-  jest.isolateModules(() => {
-    main = require('./main');
+  test('Click on treebtn without adding a conll file', async () => {
+    const expectedMessage = "Please select a ConllU/X file, or use use the Upload button in the sentence uploader section.";
+    // attach mocked event handler
+    const dialogHandler = jest.fn(dialog => dialog.dismiss());
+    page.on('dialog', dialogHandler);
+    // simulate button click
+    await page.click('#treebtn');
+    // get the dialog message and assert
+    const [firstCall] = dialogHandler.mock.calls;
+    const [dialog] = firstCall;
+    expect(dialog.message()).toEqual(expectedMessage);
   });
-});
 
+  test('Click on treebtn without adding a conll file and a config file', async () => {
```

Comment:
The goal of this test is to upload a config file, but not a conll file, and we expect it to show the dialogue message.